tensorflow/python/ops/nn_ops.py

```diff
@@ -4012,9 +4012,9 @@ def _get_noise_shape(x, noise_shape):
 
   if x.shape.dims is not None and len(x.shape.dims) == len(noise_shape_.dims):
     new_dims = []
-    for i, dim in enumerate(x.shape.dims):
-      if noise_shape_.dims[i].value is None and dim.value is not None:
-        new_dims.append(dim.value)
+    for i, dim in enumerate(tf.shape(x)):
```

Comment:
Did you run this code at all? The call to enumerate() here would raise an exception in non-autograph graph mode, and the call to numpy() below will also fail in graph mode.

Reply:
Whoops, you're right. I messed up the test implementation, and basically skipped over testing Dropout in any form completely. Made the fixes, and tested in both Graph and Eager modes.